web-ng/src/app/components/form-field-editor/form-field-editor.component.ts

```diff
@@ -219,4 +227,24 @@ export class FormFieldEditorComponent implements OnInit, OnChanges {
     options = options?.set(index, option);
     this.formOptions = new MultipleChoice(cardinality, options);
   }
+
+  drop(event: CdkDragDrop<string[]>) {
+    if (!this.formOptions) return;
+    let options = this.formOptions.options;
+    const optionAtPrevIndex = options.get(event.previousIndex);
+    const optionAtCurrentIndex = options.get(event.currentIndex);
+    if (optionAtPrevIndex && optionAtCurrentIndex) {
+      options = options.set(event.previousIndex, optionAtCurrentIndex);
+      options = options.set(event.currentIndex, optionAtPrevIndex);
+    }
+    const cardinality =
+      this.formOptions?.cardinality || Cardinality.SELECT_MULTIPLE;
+    this.formOptions = new MultipleChoice(cardinality, options);
+    this.update.emit({
```

Comment:
Looks like we rebuild the Option object and emit in three places in this file now; can we refactor that into a single function?